charts/pawtograder/images/edge-functions/main.ts

```
@@ -71,29 +71,276 @@ const MAX_RETIRED_RETRIES = 5;
+ * Concurrent misses for the same name each reserve, which over-reserves for one
+ * shared buffer. That errs toward waiting rather than toward exceeding the budget,
+ * which is the direction to err in.
+ */
+async function withEszip<T>(
+  name: string,
+  use: (eszip: Uint8Array | null) => Promise<T>
+): Promise<T> {
+  if (resident.has(name)) {
+    pin(name);
+    try {
+      return await use(await loadEszip(name));
```

ESLint

[error] React Hook "use" is called in function "withEszip" that is neither a React function component nor a custom React Hook function. React component names must start with an uppercase letter. React Hook names must start with the word "use".

(react-hooks/rules-of-hooks)